### PR TITLE
feat: add support for nwc

### LINKED ISSUE #44: Add NWC option
https://github.com/swentel/nostr-php/tree/main

### PR DESCRIPTION
**Pull request by im-adithya**
Fixes #44 

## Screenshot
Screenshot 2024-06-26 at 11 05 36 PM

### CODE REVIEW

File: admin/settings/class-connections.php [hunk 1]
```diff
@@ -73,14 +73,15 @@ public function set_form_fields()
...
-      'tab'     => 'alby',
+      'tab'     => 'nwc',
       'field'   => array(
-        'name'        => 'alby_access_token',
-        'label'       => __('Access Token', 'lnp-alby'),
-        'description' => __('Get your Alby access token with permissions for "invoices:create", "invoices:read", and "account:read" permissions from: https://getalby.com/developer/access_tokens/new', 'lnp-alby'),
+        'type'        => 'password',
+        'name'        => 'nwc_connection_uri',
+        'label'       => __('NWC Connection URI', 'lnp-alby'),
+        'description' => __('Get your NWC connection uri with permissions for "make_invoice", "lookup_invoice", "get_balance" and "get_info" permissions from: https://nwc.getalby.com/apps/new', 'lnp-alby'),
```
Comment (bumi) [inline admin/settings/class-connections.php:84]:
```suggestion
        'description' => __('Get your NWC connection uri with permissions for "make_invoice", "lookup_invoice", "get_balance" and "get_info" permissions', 'lnp-alby'),
```

File: composer.json [hunk 2]
```diff
@@ -5,13 +5,13 @@
...
-    "alby/php": "dev-main",
+    "getalby/nwc-php": "dev-main",
     "octan/lnd-rest": "dev-master",
     "octan/lnbits": "dev-master",
     "octan/lndhub": "dev-main",
     "octan/btcpay": "dev-main",
-    "firebase/php-jwt": "dev-main",
-    "jorijn/bitcoin-bolt11": "^0.1.0",
+    "firebase/php-jwt": "6.10.0",
+    "getalby/bitcoin-bolt11": "dev-master",
```
Comment (bumi) [inline composer.json:14]: why did we need to fork this?

**Review by bumi — APPROVED** [review]

File: composer.json (hunk 2 above)
Comment (im-adithya) [reply composer.json:14]: Thanks! It's not needed now, but we did when we planned to use the nostr-php library. Removing!

File: admin/settings/class-abstract-settings.php [hunk 3]
```diff
@@ -308,7 +308,11 @@ public function do_tabs_settings_section()
             );
 
             if (! empty($args['description']) ) {
-                echo esc_html($args['description']);
+                echo '<p>' . $args['description'] . '</p>';
```
Comment (bumi) [inline admin/settings/class-abstract-settings.php:311]: why did you remove `esc_html`? this is a security issue

**Review by bumi — COMMENTED** [review]
hmm, does this break things for current Alby users with the access token?

**Review by bumi — CHANGES_REQUESTED** [review]
let's keep the OAuth Alby API and add NWC separately. 
and the rest is marked as deprecated and removed in a future version.

then nothing breaks for users. 
They can connect NWC
and all options except the Alby OAuth API and NWC will be removed in a future version as they are no longer needed and all support NWC

File: admin/settings/class-abstract-settings.php (hunk 3 above)
Comment (im-adithya) [reply admin/settings/class-abstract-settings.php:311]: Didn't know that, adding!

**Comment by im-adithya** [comment]
Done!
